### PR TITLE
WDPD-209 Vault table integration tests

### PR DESCRIPTION
**Pull request by ps-cgrach**
### This PR

Create integration tests for the `Vault`

### Jira Ticket
https://jira.parkside.at/browse/WDPD-209

### CODE REVIEW

File: Tests/Integration/Core/VaultTest.php [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            ->willReturn($sToken);
+        $oSuccessResponse->method('getMaskedAccountNumber')
+            ->willReturn('Masked Account Number');
+
+        $aCard = [
+            'expiration-month' => 9,
+            'expiration-year' => 21,
+        ];
+
+        $oUser = $this->getMockBuilder(\OxidEsales\EshopCommunity\Application\Model\User::class)
```
Comment (ps-holler) [inline Tests/Integration/Core/VaultTest.php]: This would be more readable if the dependency was imported at the top via `use` (applies to the whole file).

Comment (tomazpu) [reply Tests/Integration/Core/VaultTest.php]: +1

File: Tests/Integration/Core/VaultTest.php [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                'User ID 5',
+                'Address ID 5',
+                [],
+            ],
+        ];
+    }
+
+    public function testDeleteCard()
+    {
+        $oUser = $this->getMockBuilder(\OxidEsales\EshopCommunity\Application\Model\User::class)
```
Comment (tomazpu) [inline Tests/Integration/Core/VaultTest.php]: Same here

File: Tests/Integration/Core/VaultTest.php [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $oUser->method('getId')
+            ->willReturn('User ID 2');
+
+        $this->getSession()->setUser($oUser);
+        Vault::deleteCard(4);
+        $aResult = $this->getDb(DatabaseInterface::FETCH_MODE_ASSOC)->getAll(
```
Comment (tomazpu) [inline Tests/Integration/Core/VaultTest.php]: create a private method for getting the cards

File: Tests/Integration/Core/VaultTest.php [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        $oUser->method('getId')
+            ->willReturn($sUserId);
+        $oUser->method('getSelectedAddressId')
+            ->willReturn($sAddressId);
+
+        $this->getSession()->setUser($oUser);
+
+        Vault::saveCard($oSuccessResponse, $aCard);
+
+        $aResult = $this->getDb(DatabaseInterface::FETCH_MODE_ASSOC)->getAll(
```
Comment (tomazpu) [inline Tests/Integration/Core/VaultTest.php]: create a private method for this

File: Tests/Integration/Core/VaultTest.php (hunk 1 above)
Comment (ps-cgrach) [reply Tests/Integration/Core/VaultTest.php]: 👍

File: Tests/Integration/Core/VaultTest.php (hunk 2 above)
Comment (ps-cgrach) [reply Tests/Integration/Core/VaultTest.php]: 👍

File: Tests/Integration/Core/VaultTest.php (hunk 3 above)
Comment (ps-cgrach) [reply Tests/Integration/Core/VaultTest.php]: 👍

File: Tests/Integration/Core/VaultTest.php (hunk 4 above)
Comment (ps-cgrach) [reply Tests/Integration/Core/VaultTest.php]: 👍

**Pull request merged**